### PR TITLE
Stop duplicating custom metrics per requirement

### PR DESCRIPTION
**Pull request by harry-rhesis**
## Purpose

When a test run uses custom metrics (execution-time or test-set scoped, not tied to a requirement), the Requirements performance table showed the same metric duplicated once per requirement in the run -- e.g. the same "Polish Language Detection" metric and the same failures, repeated identically under "Reliability", "Compliance", and "Robustness" -- because those metrics aren't actually specific to any requirement.

## What Changed

- **Backend** (`metric_plan.py`): `build_metric_plan` resolves a run's metrics per requirement group and previously gave every group its own payload entry unconditionally. That's correct when a group's metrics genuinely come from its own requirement (two requirements can have different metrics), but execution-time and test-set metrics apply uniformly to the whole run regardless of requirement, so every group fell through to the identical metric set and got its own duplicate copy of it. Only a requirement-sourced group now keeps its own entry; everything else (test_set, execution_time, none) pools into one requirement-less section spanning the union of those tests.
- **Frontend** (`RequirementGroup.tsx` / `RequirementTable.tsx`): the pooled, requirement-less section no longer gets a fake "requirement" header (name, drilldown icon, rollup strip labeled "Unassigned") -- these metrics were never linked to a requirement, so presenting them as if they were one was misleading. `RequirementGroup` gains a `headerless` mode that renders just the metric rows; `RequirementTable` uses it whenever a group's `id` is `null`, which after the backend fix is exactly (and only) the requirement-less bucket.

## Additional Context

- `cell_keys` (how a row's verdicts are looked up) is unaffected by this change: a real metric's cell_keys entry is always keyed by the metric's own id, never a group-relative key, so pooling which payload entry a metric's row lives under doesn't change how its verdicts resolve.

## Testing

- Backend: `uv run pytest ../../tests/backend/services/test_verdict_matrix.py ../../tests/backend/jobs/test_metric_plan_dispatch.py -q` -- 22 passed, including a new test (`TestCustomMetricPoolsAcrossRequirements`) reproducing the exact bug: two requirements, neither with its own metric, both falling through to one test-set metric -- asserts one pooled group and one verdict row instead of two duplicated ones.
- Frontend: `npx tsc --noEmit`, `npx eslint`, `npm run format:check` all clean. `npx jest --testPathPatterns="test-runs"` -- 311 passed, including a new test asserting the pooled group renders its metric rows without a group header or an "Unassigned" label.
- `uvx ruff check` / `uvx ruff format --check` clean on the backend files.

### CODE REVIEW

File: apps/backend/src/rhesis/backend/jobs/execution/metric_plan.py [hunk 1]
```diff
@@ -186,22 +197,29 @@ def _build_metric_plan(
...
+                            "name": metric.name or metric.class_name,
+                            "id": str(metric.id) if metric.id else None,
+                            "ambiguous": ambiguous,
+                        }
+                        for key, metric, ambiguous in keyed
+                    ],
+                    "test_ids": group_test_ids,
+                }
+            )
+        else:
```
[the remark this answers is not in the document]
Comment (harry-rhesis) [reply apps/backend/src/rhesis/backend/jobs/execution/metric_plan.py:226]: Thanks for catching this — fixed, and added a regression test for the exact edge case.

File: apps/frontend/src/app/(protected)/test-runs/[identifier]/components/summary/RequirementGroup.tsx [hunk 2]
```diff
@@ -113,6 +119,28 @@ export default function RequirementGroup({
   // cells under it are the same shape. Numbers mode is 0 for both.
   const stripHeight = STRIP_HEIGHTS[density];
 
+  if (headerless) {
```
[the remark this answers is not in the document]
Comment (harry-rhesis) [reply apps/frontend/src/app/(protected)/test-runs/[identifier]/components/summary/RequirementGroup.tsx]: Good catch — split into RequirementGroupHeader so that work is skipped entirely for headerless groups.
